lib/AST/Type.cpp

```diff
@@ -556,18 +556,26 @@ bool TypeBase::hasTypeRepr() const {
 }
 
 bool TypeBase::isVoid() {
-  if (auto TT = getAs<TupleType>())
-    return TT->getNumElements() == 0;
+  if (auto tuple = dyn_cast<TupleType>(this)) {
```

Comment:
This should continue to look through type sugar (i.e. the previous implementation was correct).